diff --git a/dist/index.d.ts b/dist/index.d.ts
index b7624e2500d12871f5fd10919fde03212ff761ba..58971dcdef4f4b2a1944cb1253addde055dc1562 100644
--- a/dist/index.d.ts
+++ b/dist/index.d.ts
@@ -1,3 +1,4 @@
+type AfterLogFn = ((log: Log) => void) | undefined;
 declare function getGlobalLogLevel(): LogLevel;
 declare function setGlobalLogLevel(logLevel: LogLevel): void;
 declare function getGlobalLogLevelString(): LogLevelString;
@@ -6,6 +7,8 @@ declare function getGlobalFormat(): Format;
 declare function setGlobalFormat(format: Format): void;
 declare function setGlobalTimeFormatter(fn: (inputDate: Date) => string): void;
 declare function getGlobalTimeFormatter(): (inputDate: Date) => string;
+declare function setGlobalAfterLog(fn: AfterLogFn): void;
+declare function getGlobalAfterLog(): AfterLogFn;
 interface Logg {
     /**
      *
@@ -148,6 +151,11 @@ interface Logg {
      * @param fn - callback function that takes an Error or unknown and returns an Error or unknown
      */
     withErrorProcessor: (fn: (err: Error | unknown) => Error | unknown) => Logg;
+    /**
+     * Set the after log function to a custom function
+     * @param fn - callback function that takes a Log object and returns a Log object
+     */
+    withAfterLog: (fn: (log: Log) => void) => void;
 }
 declare function createLogg(context: string): Logg;
 declare const useLogg: typeof createLogg;
@@ -203,5 +211,5 @@ declare const logLevelToColorMap: Record<LogLevel, PicoColorsFormatter>;
 declare const availableLogLevels: LogLevel[];
 declare const availableFormats: Format[];
 
-export { Format, LogLevel, LogLevelString, Format as LoggerFormat, LogLevel as LoggerLevel, availableFormats, availableLogLevelStrings, availableLogLevels, createLogg, createLogger, getGlobalFormat, getGlobalLogLevel, getGlobalLogLevelString, getGlobalTimeFormatter, initLogger, logLevelStringToLogLevelMap, logLevelToColorMap, logLevelToLogLevelStringMap, setGlobalFormat, setGlobalLogLevel, setGlobalLogLevelString, setGlobalTimeFormatter, useGlobalLogger, useLogg, useLogger };
+export { Format, LogLevel, LogLevelString, Format as LoggerFormat, LogLevel as LoggerLevel, availableFormats, availableLogLevelStrings, availableLogLevels, createLogg, createLogger, getGlobalAfterLog, getGlobalFormat, getGlobalLogLevel, getGlobalLogLevelString, getGlobalTimeFormatter, initLogger, logLevelStringToLogLevelMap, logLevelToColorMap, logLevelToLogLevelStringMap, setGlobalAfterLog, setGlobalFormat, setGlobalLogLevel, setGlobalLogLevelString, setGlobalTimeFormatter, useGlobalLogger, useLogg, useLogger };
 export type { Log, Logg, Logger, LoggerConfig };
diff --git a/dist/index.mjs b/dist/index.mjs
index 922d49fead21ddba4af149c9c9eda6d8e3197b95..df72de15f955b28f7eee978cf4c48e46e77ff135 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -298,7 +298,8 @@ const GLOBAL_CONFIG = {
   configured: false,
   logLevel: LogLevel.Debug,
   format: Format.JSON,
-  timeFormatter: (inputDate) => inputDate.toISOString()
+  timeFormatter: (inputDate) => inputDate.toISOString(),
+  afterLog: void 0
 };
 function getGlobalLogLevel() {
   return GLOBAL_CONFIG.logLevel;
@@ -352,6 +353,13 @@ function setGlobalTimeFormatter(fn) {
 function getGlobalTimeFormatter() {
   return GLOBAL_CONFIG.timeFormatter;
 }
+function setGlobalAfterLog(fn) {
+  GLOBAL_CONFIG.afterLog = fn;
+  GLOBAL_CONFIG.configured = true;
+}
+function getGlobalAfterLog() {
+  return GLOBAL_CONFIG.afterLog;
+}
 function createLogg(context) {
   const logObj = {
     fields: {},
@@ -360,6 +368,7 @@ function createLogg(context) {
     format: Format.JSON,
     shouldUseGlobalConfig: false,
     errorProcessor: (err) => err,
+    afterLog: void 0,
     timeFormatter: (inputDate) => inputDate.toISOString(),
     useGlobalConfig: () => {
       logObj.shouldUseGlobalConfig = true;
@@ -508,6 +517,9 @@ function createLogg(context) {
       const logger = logObj.child();
       logger.errorProcessor = fn;
       return logger;
+    },
+    withAfterLog: (fn) => {
+      logObj.afterLog = fn;
     }
   };
   const getEffectiveLogLevel = () => {
@@ -552,6 +564,10 @@ function createLogg(context) {
     }
     const output = format === Format.Pretty ? toPrettyString(raw) : JSON.stringify(raw);
     console[consoleMethod](output);
+    const afterLogFn = logObj.afterLog ?? getGlobalAfterLog();
+    if (afterLogFn != null) {
+      afterLogFn(raw);
+    }
   };
   const logLevelCheckMap = {
     [LogLevelString.Debug]: shouldOutputDebugLevelLogWhenLogLevelIsOneOf,
@@ -632,4 +648,4 @@ function initLogger(level = LogLevel.Verbose, format = Format.Pretty) {
   logger.withFields({ level: LogLevel[level], format }).log("Logger initialized");
 }
 
-export { Format, LogLevel, LogLevelString, Format as LoggerFormat, LogLevel as LoggerLevel, availableFormats, availableLogLevelStrings, availableLogLevels, createLogg, createLogger, getGlobalFormat, getGlobalLogLevel, getGlobalLogLevelString, getGlobalTimeFormatter, initLogger, logLevelStringToLogLevelMap, logLevelToColorMap, logLevelToLogLevelStringMap, setGlobalFormat, setGlobalLogLevel, setGlobalLogLevelString, setGlobalTimeFormatter, useGlobalLogger, useLogg, useLogger };
+export { Format, LogLevel, LogLevelString, Format as LoggerFormat, LogLevel as LoggerLevel, availableFormats, availableLogLevelStrings, availableLogLevels, createLogg, createLogger, getGlobalAfterLog, getGlobalFormat, getGlobalLogLevel, getGlobalLogLevelString, getGlobalTimeFormatter, initLogger, logLevelStringToLogLevelMap, logLevelToColorMap, logLevelToLogLevelStringMap, setGlobalAfterLog, setGlobalFormat, setGlobalLogLevel, setGlobalLogLevelString, setGlobalTimeFormatter, useGlobalLogger, useLogg, useLogger };
